src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -111,7 +111,10 @@ public override void VisitRazorDirective(RazorDirectiveSyntax node)
 
         public override void VisitRazorDirectiveBody(RazorDirectiveBodySyntax node)
         {
-            AddSemanticRange(node.Keyword, SyntaxKind.RazorDirective);
+            if (node.Keyword.Kind != SyntaxKind.CSharpStatementLiteral)
```

Comment:
```suggestion
            // We can't provide colors for CSharp because if we both provided them then they would overlap, which violates the LSP spec.
            if (node.Keyword.Kind != SyntaxKind.CSharpStatementLiteral)
```